services/src/main/java/com/broadleafcommerce/bulkoperations/web/endpoint/BulkOperationsEndpoint.java

```diff
@@ -0,0 +1,84 @@
+import org.broadleafcommerce.frameworkmapping.annotation.FrameworkPostMapping;
+import org.broadleafcommerce.frameworkmapping.annotation.FrameworkRestController;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import com.broadleafcommerce.bulk.v2.domain.BulkOperationRequest;
+import com.broadleafcommerce.bulk.v2.domain.BulkOperationResponse;
+import com.broadleafcommerce.bulkoperations.service.BulkOperationsService;
+import com.broadleafcommerce.bulkoperations.service.handler.BulkOperationHandler;
+import com.broadleafcommerce.bulkoperations.web.exception.BulkOperationHandlerNotFoundException;
+import com.broadleafcommerce.data.tracking.core.context.ContextInfo;
+import com.broadleafcommerce.data.tracking.core.context.ContextOperation;
+import com.broadleafcommerce.data.tracking.core.policy.Policy;
+import com.broadleafcommerce.data.tracking.core.type.OperationType;
+
+import java.util.List;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@FrameworkRestController
+@FrameworkMapping(BASE_URI)
+@RequiredArgsConstructor
+@Slf4j
+public class BulkOperationsEndpoint {
+
+    public static final String BASE_URI = "/bulk-operations";
+
+    @Getter(AccessLevel.PROTECTED)
+    private final BulkOperationsService bulkOperationsService;
+
+    @Getter(AccessLevel.PROTECTED)
+    private final List<BulkOperationHandler> bulkOperationHandlers;
+
+    @FrameworkPostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Policy(permissionRoots = "BULK_OPERATION")
+    public BulkOperationResponse createBulkOperation(HttpServletRequest request,
+            @ContextOperation(value = OperationType.CREATE) ContextInfo context,
+            @RequestBody BulkOperationRequest bulkOperationRequest) {
+        bulkOperationsService.validateBulkOperationRequest(bulkOperationRequest, context);
+
+        for (BulkOperationHandler bulkOperationHandler : bulkOperationHandlers) {
+            if (bulkOperationHandler.canHandle(bulkOperationRequest.getOperationType(),
+                    bulkOperationRequest.getEntityType())) {
+                return bulkOperationHandler.handle(bulkOperationRequest, context);
+            } else {
+                log.warn("No handler was found for operation type {} and entity type {}",
+                        bulkOperationRequest.getOperationType(),
+                        bulkOperationRequest.getEntityType());
+
+                throw new BulkOperationHandlerNotFoundException(
+                        String.format(
+                                "No handler was found for operation type %s and entity type %s",
+                                bulkOperationRequest.getOperationType(),
+                                bulkOperationRequest.getEntityType()));
+            }
+        }
+        return null;
```

Comment:
I think you meant this, otherwise the exception gets thrown if one of the handlers can't handle the operation.

Not sure about throwing an exception and logging a warning. Maybe log an error instead or warning?

```suggestion
        for (BulkOperationHandler bulkOperationHandler : bulkOperationHandlers) {
            if (bulkOperationHandler.canHandle(bulkOperationRequest.getOperationType(),
                    bulkOperationRequest.getEntityType())) {
                return bulkOperationHandler.handle(bulkOperationRequest, context);
            }
        }

        log.warn("No handler was found for operation type {} and entity type {}",
                bulkOperationRequest.getOperationType(),
                bulkOperationRequest.getEntityType());

        throw new BulkOperationHandlerNotFoundException(
                String.format(
                        "No handler was found for operation type %s and entity type %s",
                        bulkOperationRequest.getOperationType(),
                        bulkOperationRequest.getEntityType()));
```